Sources/AWSSDKSwiftCore/AWSClient.swift

```diff
@@ -413,30 +421,74 @@ extension AWSClient {
         httpMethod: HTTPMethod,
         serviceConfig: AWSServiceConfig,
         input: Input,
+        context: AWSClient.Context,
         on eventLoop: EventLoop? = nil,
-        logger: Logger = AWSClient.loggingDisabled,
         stream: @escaping AWSHTTPClient.ResponseStream
+    ) -> EventLoopFuture<Output> {
+        return execute(
+            operation: operationName,
+            createRequest: { _ in
+                try AWSRequest(
+                    operation: operationName,
+                    path: path,
+                    httpMethod: httpMethod,
+                    input: input,
+                    configuration: serviceConfig
+                )
+            },
+            execute: { request, eventLoop, context in
+                return self.httpClient.execute(request: request, timeout: serviceConfig.timeout, on: eventLoop, context: context, stream: stream)
+            },
+            processResponse: { response in
+                return try self.validate(operation: operationName, response: response, serviceConfig: serviceConfig)
+            },
+            config: serviceConfig,
+            context: context,
+            on: eventLoop
+        )
+    }
+
+    /// internal version of execute
+    internal func execute<Output>(
+        operation operationName: String,
+        createRequest: @escaping (AWSSigner) throws -> AWSRequest,
+        execute: @escaping (AWSHTTPRequest, EventLoop, AWSClient.Context) -> EventLoopFuture<AWSHTTPResponse>,
+        processResponse: @escaping (AWSHTTPResponse) throws -> Output,
+        config: AWSServiceConfig,
+        context: AWSClient.Context,
+        on eventLoop: EventLoop? = nil
     ) -> EventLoopFuture<Output> {
         let eventLoop = eventLoop ?? eventLoopGroup.next()
-        let logger = logger.attachingRequestId(Self.globalRequestID.add(1), operation: operationName, service: serviceConfig.service)
-        let future: EventLoopFuture<Output> = credentialProvider.getCredential(on: eventLoop, logger: logger).flatMapThrowing { credential in
-            let signer = AWSSigner(credentials: credential, name: serviceConfig.signingName, region: serviceConfig.region.rawValue)
-            let awsRequest = try AWSRequest(
-                operation: operationName,
-                path: path,
-                httpMethod: httpMethod,
-                input: input,
-                configuration: serviceConfig
-            )
+        // TODO: discuss hot to update the context (and its baggage)
+        var context = context
+        context.setRequest(.init(operation: operationName, serviceConfig: config))
+        let span = InstrumentationSystem.tracingInstrument.startSpan(
+            named: "\(config.service):\(operationName)", // TODO: or just "execute"?
+            context: context
+        )
+        context = context.with(baggage: span.context)
```

Comment:
@ktoso this is my main question to you, see also https://github.com/slashmo/gsoc-swift-tracing/issues/126#issuecomment-675294268

what's the recommended way to update and propagate (baggage) context, 

`TracingInstrument` updates only `BaggageContext` (updated value is provided in span `context` attribute),
this was straightforward until `BaggageContextCarrier`

I created a convenience function as above, but its not great, one of the main problems is that it replaces baggage;

on a side note `NoOpSpan` as of now returns always empty `BaggageContext`,
this is a problem since its no longer used only for tracing but also to propagate logger metadata (which is wiped by `NoOpSpan` using the approach above)

Reply:
Is this the cause of the lost `Logger` metadata?

Reply:
> Is this the cause of the lost Logger metadata?

yes, see comment above https://github.com/swift-aws/aws-sdk-swift-core/pull/348#discussion_r479000366

Reply:
fixed with https://github.com/slashmo/gsoc-swift-tracing/issues/138